osvr/RenderKit/RenderManagerD3DOpenGL.cpp

```diff
@@ -444,94 +417,126 @@ namespace renderkit {
         }
         if (oglMap == nullptr) {
-            std::cerr
-                << "RenderManagerD3D11OpenGL::PresentEye(): Unregistered buffer"
-                << " (call RegisterRenderBuffers before presenting)"
-                << std::endl;
-            return false;
+          std::cerr
+            << "RenderManagerD3D11OpenGL::PresentRenderBuffersInternal(): Unregistered buffer"
+            << " (call RegisterRenderBuffers before presenting)"
+            << std::endl;
+          return false;
         }
-        if (params.m_buffer.OpenGL->colorBufferName != oglMap->OpenGLTexture) {
-            std::cerr
-                << "RenderManagerD3D11OpenGL::PresentEye(): Mis-matched buffer"
-                << " (call RegisterRenderBuffers whenever a new render-texture "
-                   "is created)"
-                << std::endl;
-            return false;
+        if (renderBuffers[b].OpenGL->colorBufferName != oglMap->OpenGLTexture) {
+          std::cerr
+            << "RenderManagerD3D11OpenGL::PresentRenderBuffersInternal(): Mis-matched buffer"
+            << " (call RegisterRenderBuffers whenever a new render-texture "
+            "is created)"
+            << std::endl;
+          return false;
         }
-
-        // Create a new param buffer and fill in the new D3D buffer pointer
-        // along with our current info.  Invert the flip-buffer flag, because
-        // we need to render upside-down.  Rotate around the negative Z axis
-        // rather than the positive.  Then call the D3D PresentEye method
-        // and clean up after ourselves.
-        PresentEyeParameters sendParams = params;
-        sendParams.m_flipInY = !sendParams.m_flipInY;
-        sendParams.m_buffer.OpenGL = nullptr;
-        sendParams.m_buffer.D3D11 = new RenderBufferD3D11;
-        sendParams.m_buffer.D3D11->colorBuffer = oglMap->D3DTexture;
-        sendParams.m_buffer.D3D11->colorBufferView =
-            oglMap->D3DrenderTargetView;
-
-        // Transpose the texture matrix because we're feeding it to Direct3D,
-        // which stores matrix elements in a different order.  We're sending
-        // it our OpenGL transforms because we used OpenGL to render the two
-        // images, so that's what the texture matrix needs to assume.
-        matrix16 textureMat;
-        if (params.m_ATW != nullptr) {
-            // Transpose the matrix, because it is in OpenGL format.
-            for (size_t r = 0; r < 4; r++) {
-                for (size_t c = 0; c < 4; c++) {
-                    textureMat.data[r * 4 + c] = params.m_ATW->data[c * 4 + r];
-                }
-            }
-            sendParams.m_ATW = &textureMat;
+        RenderBuffer rb;
+        rb.D3D11 = &(oglMap->D3DBuffer);
+        myD3DBuffers.push_back(rb);
+      }
+
+      // Unlock all of the render buffers we know about.
+      for (size_t i = 0; i < m_oglToD3D.size(); i++) {
+        if (!wglDXUnlockObjectsNV(m_glD3DHandle, 1, &m_oglToD3D[i].glColorHandle)) {
```

Comment:
NVM, looked it up. http://halogenica.net/sharing-resources-between-directx-and-opengl/